File: Seccion 11 - Funciones en C++/Seccion_11_Python/17_ejercicio_14.py

vector = []
tam = 0

def pedir_datos():
    global vector, tam
    
    tam = int(input('Digite el tamaño del vector: '))
    
    for i in range(tam):
        valor = int(input('{}. Digite un número: '.format(i + 1)))
        vector.append(valor)

def impares():
    vec_impares = []
    
    for valor in vector:
        if valor % 2 != 0:
            vec_impares.append(valor)
    
    print('\nImprimiendo los elementos impares del vector:\n')
    print(vec_impares)

pedir_datos()
impares()

File: Seccion 11 - Funciones en C++/Seccion_11_Dart/17_ejercicio_14.dart

import 'dart:io';

List<int> vector = new List();
int tam;

void pedirDatos() {
  int valor;

  stdout.writeln('Digite el tamaño del vector: ');
  tam = int.parse(stdin.readLineSync());

  for (int i = 0; i < tam; i++) {
    stdout.writeln('${i +1}. Digite un número: ');
    valor = int.parse(stdin.readLineSync());
    vector.insert(i, valor);
  }
}

void impares() {
  List<int> vecImpares = new List();
  int j = 0;

  for (int i = 0; i < tam; i++) {
    if (vector[i] % 2 != 0) {
      vecImpares.insert(j, vector[i]);
      j++;
    }
  }

  print('\nImprimiendo los elementos impares del vector:\n');
  print(vecImpares);
}

void main() {

  pedirDatos();
  impares();
}